lace_ground.py

```diff
@@ -13,215 +13,274 @@
-		
-		result = self.loadFile(self.fname)
-		
-		#Convert input from mm to pixels, assuming 90 dpi
-		conversion = 90.0 / 25.4
-		self.options.width *= conversion
-		self.options.height *= conversion
-		self.options.size *= conversion
-		
-		# Users expect spacing to be the vertical distance between footside pins (vertical distance between every other row) 
-		# but in the script we use it as as diagonal distance between grid points
-		# therefore convert spacing based on the angle chosen
-		theta = radians(self.options.angle)
-		self.options.spacing *= conversion/(2.0*cos(theta))
-		
-		# Convert color from long integer to hexidecimal string
-		self.options.color = self.getColorString(self.options.color)
-		
-		# Draw a ground based on file description and user inputs
-		# For now, assume style is Checker but could change in future
-		self.drawCheckerGround(result["data"],result["rowCount"],result["colCount"])
+        p1, p2, p3 = pairs
+        #inkex.debug("%s"%(pairs))
+        path = "M %s,%s L %s,%s M %s,%s L %s,%s" %(p1[0], p1[1], p2[0], p2[1], p1[0], p1[1], p3[0], p3[1])
+        
+        # define the stroke style
+        s = {'stroke-linejoin': 'miter', 
+            'stroke-width': '0.5px',
+            'stroke': '#000000', 
+            'stroke-linecap': 'butt',
+            # 'marker-start':'url(#Arrow1Lend)', # no useful style - need to make manually
+            # 'stroke-opacity': '1.0', 
+            # 'fill-opacity': '1.0',
+            'fill': 'none'
+        }
+        # create attributes from style and path
+        attribs = {'style':simplestyle.formatStyle(s), 'd':path}
+        # insert path object into current layer
+        inkex.etree.SubElement(parent, inkex.addNS('path', 'svg'), attribs)
+
+    def circle(self, x, y, r, fill, stroke, strokeWidth, parent):
+        """
+        Draw a circle of radius 'r' and origin at (x, y)
+        """
+        # define the stroke style
+        style = {'stroke-width': strokeWidth,
+                 'stroke': stroke,
+                 'stroke-linecap': 'butt',
+                 'stroke-linejoin': 'miter',
+                 # 'fill-opacity': '1.0',
+                 # 'stroke-opacity': '1.0',
+                 'fill': fill
+                }
+        # create attributes from style and define path
+        attribs = {'style':simplestyle.formatStyle(style), 
+                   'cx':str(x),
+                   'cy':str(y),
+                   'r':str(r)
+                  }
+        # insert path object into current layer
+        inkex.etree.SubElement(parent, inkex.addNS('circle', 'svg'), attribs)
+
+    def draw_grid_pin(self, x, y, parent):
+        " Draw a single grid pin "
+        pin_radius = self.options.pindia/2
+        fill = self.options.pincolor
+        self.circle(x, y, pin_radius, fill, '#000000',  '0.1mm', parent)
+
+    def loadFile(self, fname):
+        data = []
+        rowCount = 0
+        colCount = 0
+        with open(fname,'r') as f:

```

Comment:
This could be improved by using 'rU' for both kinds of line endings. but may be deprecated so leaving for now.